packages/backend/src/plugins/DemoEventBasedEntityProvider.ts

```diff
@@ -18,42 +18,40 @@ import {
   EntityProvider,
   EntityProviderConnection,
 } from '@backstage/plugin-catalog-node';
-import {
-  EventBroker,
-  EventParams,
-  EventSubscriber,
-} from '@backstage/plugin-events-node';
+import { EventParams, EventService } from '@backstage/plugin-events-node';
 import { Logger } from 'winston';
 
-export class DemoEventBasedEntityProvider
-  implements EntityProvider, EventSubscriber
-{
+export class DemoEventBasedEntityProvider implements EntityProvider {
   private readonly logger: Logger;
+  private readonly eventService: EventService;
   private readonly topics: string[];
 
   constructor(opts: {
-    eventBroker: EventBroker;
+    eventService: EventService;
     logger: Logger;
     topics: string[];
   }) {
-    const { eventBroker, logger, topics } = opts;
-    this.logger = logger;
-    this.topics = topics;
-    eventBroker.subscribe(this);
+    this.eventService = opts.eventService;
+    this.logger = opts.logger;
+    this.topics = opts.topics;
+  }
+
+  async subscribe() {
+    await this.eventService.subscribe({
+      id: this.constructor.name,
```

Comment:
This isn't very safe because of minification and the potential for conflicting names and differences across builds. I think it's best if we encourage something different.

Reply:
I can use a fixed string instead like
```suggestion
      id: 'DemoEventBasedEntityProvider',
```